codes/ruby/chapter_backtracking/preorder_traversal_i_compact.rb

```diff
@@ -0,0 +1,33 @@
+=begin
+File: preorder_traversal_i_compact.rb
+Created Time: 2024-04-07
+Author: hoangtien2k3 (hoangtien2k3qx1@gmail.com)
+=end
+
+require_relative 'utils'
+
+# 前序遍历：例题一
+def pre_order(root)
+  return if root.nil?
+
+  if root.val == 7
+    # 记录解
+    $res << root
+  end
+
+  pre_order(root.left)
+  pre_order(root.right)
+end
+
+root = TreeNode.list_to_tree([1, 7, 3, 4, 5, 6, 7])
```

Comment:
There is no `TreeNode#list_to_tree` method, please transpile the corresponding method from the `tree_node.py`. Keep in mind:
- The name should change all the `list` to `array`. Because in Ruby we tend to call it "arr".
- For the consistency, the method should not be an instance method of `TreeNode` class.

Reply:
@hoangtien2k3 It seems you didn't test the code before submitting the PR. The code in this book is for beginners, so please carefully check and test the code.